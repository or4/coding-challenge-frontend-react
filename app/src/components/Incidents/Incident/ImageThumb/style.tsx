import styled, { StyledComponent } from 'styled-components';
import { media } from 'ui/utils';

import prevewSvg from './assets/preview.svg';
import { IProps } from '.';

interface IContainerProps {
    isPreview: boolean;
}

export const Container: StyledComponent<'div', any, IContainerProps, never> = styled.div`
    ${media.mobile`
        display: ${(props: IContainerProps) => (props.isPreview ? 'none' : 'inherit')};
		padding-bottom: 10px;
  	`}

    ${media.tabletNDesktop`
		width: 100px;
		height: 100px;
		min-width: 100px;
		min-height: 100px;

		display: flex;
		align-items: center;
		justify-content: center;
		box-sizing: border-box;
		border-radius: 4px;
		border: 1px solid rgba(0, 0, 0, 0.05);
    	background: rgba(0, 0, 0, 0.02);
	`}
`;

export const Image = styled.div`
    background-image: url(${(props: IProps) => props.imageUrl});
    background-repeat: no-repeat;
    background-size: cover;

    ${media.mobile`
		padding-top: 100%;
		width: 100%;
  	`}

    ${media.tabletNDesktop`
		width: 90px;
		height: 90px;
	`}
`;

export const Preview = styled.div`
    background-image: url(${prevewSvg});
    background-repeat: no-repeat;
    background-size: cover;

    width: 55px;
    height: 32px;

    ${media.mobile`
		display: none;
  	`}
`;
