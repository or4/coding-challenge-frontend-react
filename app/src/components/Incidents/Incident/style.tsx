import styled from 'styled-components';
import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    cursor: pointer;
    padding-top: 10px;
    padding-bottom: 10px;
    font-size: 14px;
    outline: none;

    ${media.mobile`
		flex-direction: column;
		border-bottom: 1px solid rgba(0, 0, 0, 0.1);
  	`}

    ${media.tabletNDesktop`
    	min-height: 100px;
	`}

    &:hover {
        background: rgba(0, 0, 0, 0.03);
    }
`;

export const TextContainer = styled.div`
    display: flex;
    flex-direction: column;

    ${media.tabletNDesktop`
        padding: 0 15px;
    `}
`;

export const IncidentTitle = styled.div`
    font-weight: 600;

    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDescription = styled.div`
    ${media.mobile`
        padding-bottom: 5px;
    `}

    ${media.tabletNDesktop`
        padding-bottom: 10px;
    `}
`;

export const IncidentDetails = styled.div``;
