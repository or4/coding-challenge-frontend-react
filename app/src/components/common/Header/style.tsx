import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    display: flex;
    align-items: center;
    cursor: pointer;

    ${media.mobile`
        height: 60px;
        margin-bottom: 15px;
    `}

    ${media.tablet`
        height: 100px;
    `}

    ${media.desktop`
        height: 140px;
    `}
`;

export const Logo = styled.img`
    width: 64px;
    height: 64px;

    ${media.tabletNDesktop`
        padding: 0 18px;
    `}
`;

export const TextWrapper = styled.div`
    padding-left: 15px;
`;

export const Title = styled.div`
    font-weight: 600;

	${media.mobile`
    	font-size: 18px;
  	`}

	${media.tablet`
    	font-size: 32px;
	`}

    ${media.desktop`
    	font-size: 48px;
	`}
`;

export const Description = styled.div`
	font-weight: 600;

	${media.mobile`
    	font-size: 14px;
  	`}

	${media.tablet`
    	font-size: 16px;
	`}

    ${media.desktop`
    	font-size: 32px;
	`}
`;
