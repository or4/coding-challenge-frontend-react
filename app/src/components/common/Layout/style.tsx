import styled from 'styled-components';

import { media } from 'ui/utils';

export const Container = styled.div`
    margin: 0 auto 15px;

    ${media.mobile`
		padding: 15px;
  	`}

	${media.tablet`
		padding: 15px;
	`}

    ${media.desktop`
		max-width: 940px;
	`}
`;
