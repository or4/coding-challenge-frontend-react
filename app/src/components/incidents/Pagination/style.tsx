import styled from 'styled-components';

import { PageButton } from 'ui/PageButton';

export const Container = styled.div`
    text-align: center;
    margin: 15px 0;
`;

export const Button = styled(PageButton)`
    margin-right: calc(15px + 3px);

    &:last-child {
        margin-right: 0;
    }
`;
