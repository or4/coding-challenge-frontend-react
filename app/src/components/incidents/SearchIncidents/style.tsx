import styled from 'styled-components';
import { media } from 'ui/utils';

export const QueryInput = styled.input`
    min-width: 240px;
    padding: 4px 8px;
    flex-grow: 1;
    margin: 5px 5px;

    ${media.mobileNTablet`
		width: 100%;
	`}
`;
