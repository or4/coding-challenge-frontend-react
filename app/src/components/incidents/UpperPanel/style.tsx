import styled from 'styled-components';

export const Container = styled.div`
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    margin: -5px -5px;
`;
