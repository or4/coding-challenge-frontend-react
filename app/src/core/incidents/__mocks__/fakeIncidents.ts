import { times } from 'lodash';
import { IIncident } from 'types';

export const getFakeIncidents: (amount: number) => IIncident[] = amount => {
    return times(amount, index => ({
        id: index + 1,
        address: 'Berlin, 10963, DE',
        description: "Stolen during nighttime from the house's closed yard.",
        locationDescription: null,
        locationType: null,
        media: {
            imageUrl: 'https://files.bikeindex.org/uploads/Pu/196548/large_Bike01.png',
            imageUrlThumb: 'https://files.bikeindex.org/uploads/Pu/196548/small_Bike01.png',
        },
        occurredAt: 1571918301,
        source: {
            name: 'BikeIndex.org',
            htmlUrl: 'https://bikeindex.org/bikes/669448',
            apiUrl: 'https://bikeindex.org/api/v1/bikes/669448',
        },
        title: 'Stolen 2017 State Bicycle Co. Matte Black 5 - 59 Core Line: Fixed Gear(black)',
        type: 'Theft',
        typeProperties: null,
        updatedAt: 1573322470,
        url: 'https://bikewise.org/api/v1/incidents/109462',
    }));
};
