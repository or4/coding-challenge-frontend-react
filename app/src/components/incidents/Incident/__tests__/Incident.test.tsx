import React from 'react';
import { shallow } from 'enzyme';
import { getFakeIncidents } from 'core/incidents/__mocks__/fakeIncidents';
import { IIncident } from 'types';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { Incident } from '../index';
import { IncidentTitle, IncidentDescription, IncidentDetails } from '../style';

describe('Incident', () => {
    let incident: IIncident;

    beforeEach(() => {
        incident = getFakeIncidents(1)[0];
    });

    it('should correct render component', () => {
        const tree = renderer.create(<Incident {...incident} />).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should have title', () => {
        const wrapper = shallow(<Incident {...incident} />);
        const title = wrapper.find(IncidentTitle);

        expect(title.exists()).toBeTruthy();
        expect(title).toHaveLength(1);
        expect(title.children().text()).toBe(incident.title);
    });

    it('should have description', () => {
        const wrapper = shallow(<Incident {...incident} />);
        const description = wrapper.find(IncidentDescription);

        expect(description.exists()).toBeTruthy();
        expect(description).toHaveLength(1);
        expect(description.children().text()).toBe(incident.description);
    });

    it('should have details', () => {
        const wrapper = shallow(<Incident {...incident} />);
        const description = wrapper.find(IncidentDetails);

        expect(description.exists()).toBeTruthy();
        expect(description).toHaveLength(1);
        expect(description.text()).toBe('Thu Oct 24 2019 - Berlin, 10963, DE');
    });

    it('should no description', () => {
        const wrapper = shallow(<Incident {...incident} description={undefined} />);
        const description = wrapper.find(IncidentDescription);

        expect(description.exists()).toBeFalsy();
        expect(description).toHaveLength(0);
    });
});
