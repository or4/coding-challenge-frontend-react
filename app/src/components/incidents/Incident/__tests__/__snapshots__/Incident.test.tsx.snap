// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incident should correct render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding-top: 10px;
  padding-bottom: 10px;
  font-size: 14px;
  outline: none;
}

.c0:hover {
  background: rgba(0,0,0,0.03);
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-weight: 600;
}

.c2 {
  background-repeat: no-repeat;
  background-size: cover;
}

@media (max-width:479px) {
  .c0 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    border-bottom: 1px solid rgba(0,0,0,0.1);
  }
}

@media (min-width:480px) {
  .c0 {
    min-height: 100px;
  }
}

@media (min-width:480px) {
  .c3 {
    padding: 0 15px;
  }
}

@media (max-width:479px) {
  .c4 {
    padding-bottom: 5px;
  }
}

@media (min-width:480px) {
  .c4 {
    padding-bottom: 10px;
  }
}

@media (max-width:479px) {
  .c5 {
    padding-bottom: 5px;
  }
}

@media (min-width:480px) {
  .c5 {
    padding-bottom: 10px;
  }
}

@media (max-width:479px) {
  .c1 {
    display: inherit;
    padding-bottom: 10px;
  }
}

@media (min-width:480px) {
  .c1 {
    width: 100px;
    height: 100px;
    min-width: 100px;
    min-height: 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    box-sizing: border-box;
    border-radius: 4px;
    border: 1px solid rgba(0,0,0,0.05);
    background: rgba(0,0,0,0.02);
  }
}

@media (max-width:479px) {
  .c2 {
    padding-top: 100%;
    width: 100%;
  }
}

@media (min-width:480px) {
  .c2 {
    width: 90px;
    height: 90px;
  }
}

<div
  className="c0"
  data-test-id="incident"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      data-test-id="incident-image__image"
      style={
        Object {
          "backgroundImage": "url(https://files.bikeindex.org/uploads/Pu/196548/small_Bike01.png)",
        }
      }
    >
      <img
        alt="incident"
        height="0"
        src="https://files.bikeindex.org/uploads/Pu/196548/small_Bike01.png"
      />
    </div>
  </div>
  <div
    className="c3"
  >
    <div
      className="c4"
      data-test-id="incident__title"
    >
      Stolen 2017 State Bicycle Co. Matte Black 5 - 59 Core Line: Fixed Gear(black)
    </div>
    <div
      className="c5"
      data-test-id="incident__description"
    >
      Stolen during nighttime from the house's closed yard.
    </div>
    <div
      className=""
    >
      <span
        data-test-id="incident__date"
      >
        Thu Oct 24 2019
      </span>
       â€“ 
      <span
        data-test-id="incident__address"
      >
        Berlin, 10963, DE
      </span>
    </div>
  </div>
</div>
`;
